src/foam/dao/AbstractDAO.js

```diff
@@ -591,7 +592,7 @@ if ( ( skip > 0 ) && ( skip < AbstractDAO.MAX_SAFE_INTEGER ) ) {
   sink = new SkipSink(skip, 0, sink);
 }
 
-if ( order != null ) {
+if ( order != null && ! isCount ) {
```

Comment:
I think your original code was correct, and now you've broken it (at my suggestion). If skip or limit were set then it would no longer be a Count because it would have been decorated with a SkipSink or LimitedSink. So what you originally had was correct.